scripts/schema.sql

```diff
@@ -17,3 +17,20 @@ CREATE TABLE IF NOT EXISTS sessions (
     FOREIGN KEY (user_id) REFERENCES users(user_id) ON DELETE CASCADE
 );
 
+CREATE TABLE IF NOT EXISTS friend_requests (
+    sender_id INT NOT NULL,
+    receiver_id INT NOT NULL,
+    sent_on TIMESTAMPTZ NOT NULL,
+    is_confirmed BOOLEAN NOT NULL DEFAULT 'no',
+    PRIMARY KEY (sender_id, receiver_id),
+    FOREIGN KEY (sender_id) REFERENCES users(user_id) ON DELETE CASCADE,
+    FOREIGN KEY (receiver_id) REFERENCES users(user_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS friends (
+    user_id_1 INT NOT NULL,
+    user_id_2 INT NOT NULL,
+    PRIMARY KEY (user_id_1, user_id_2),
+    FOREIGN KEY (user_id_1) REFERENCES users(user_id) ON DELETE CASCADE,
+    FOREIGN KEY (user_id_2) REFERENCES users(user_id) ON DELETE CASCADE
+);
```

Comment:
Ca à l'air pas mal, par contre t'es sur de toi pour les pk's dans tes tables? Je sais plus si on est sensé d'office utiliser ça quand c'est du many to many.

Reply:
Oui, car ça permet que la liaison entre le user1 et le user2 soit unique, puisque la PK se fait sur les 2 colonnes.

Reply:
Le truc c'est qu'on peut faire ça et du coup quand même avoir une pk "normal" ça me met le doute:
``` sql
UNIQUE (sender_user_id, receiver_user_id)
```

Reply:
Oui effectivement il y a plusieurs solutions possibles. En fait, j'ai d'abord écrit moi-même la création des tables, puis j'ai demandé à ChatGPT d'optimiser au maximum et j'ai obtenu ceci. Si tu préfères je veux bien changer, mais alors il faut que je rajoute un champ "id" dédié à chaque table sur lesquels j'assigne à chacun une PK classique.

Reply:
Comme tu veux, je sais pas vraiment ce qui est mieux, implémente ce que tu préfère et je validerai la pr, de toute façon les deux solutions sont bonnes à mon avis.

Reply:
Ok tu peux valider, juste je ne sais pas si on merge directement ou pas. Si oui, il ne faut pas supprimer la branche, car toutes les fonctionnalités de demandes d'ami ne sont pas encore implémentées.